app/web_hub.go

```diff
@@ -549,9 +560,10 @@ type hubConnectionIndexIndexes struct {
 
 // hubConnectionIndex provides fast addition, removal, and iteration of web connections.
 type hubConnectionIndex struct {
-	connections         []*WebConn
-	connectionsByUserId map[string][]*WebConn
-	connectionIndexes   map[*WebConn]*hubConnectionIndexIndexes
+	connections            []*WebConn
+	connectionsByUserIdMut sync.RWMutex
```

Comment:
Let me know if you think adding this mutex is not good for scalability, because the connectionIndex was local to the `Start` method before, but now we need it as a member variable.

The way around it would be to do some channel juggling to pass a msg through a dummy channel to get the connection count and then have a return channel to get back the result. This maintains performance at the cost of slightly complicated and unreadable code.

Thoughts welcome.